public/app/plugins/panel/piechart/PieChart.tsx

```diff
@@ -63,7 +63,7 @@ export const PieChart: FC<PieChartProps> = ({
   });
 
   const filteredFieldDisplayValues = fieldDisplayValues.filter((dv) => {
-    return !dv.field.custom.hideFrom.viz;
+    return !dv.field.custom.hideFrom.viz && !isNaN(dv.display.numeric);
```

Comment:
nit: could we extract a common utility function that can be reused?

Reply:
Good point, I'll fix that and also make sure we use the missing values setting instead of hard coding to -.